core/src/main/java/com/scalar/db/common/ProjectedResult.java

```diff
@@ -38,26 +37,29 @@ public ProjectedResult(Result original, List<String> projections) {
   @Deprecated
   @Override
   public Optional<Key> getPartitionKey() {
-    return getKey(original.getPartitionKey());
+    Optional<Key> partitionKey = original.getPartitionKey();
+    checkKey(partitionKey);
+    return partitionKey;
   }
 
   /** @deprecated As of release 3.8.0. Will be removed in release 5.0.0 */
   @Deprecated
   @Override
   public Optional<Key> getClusteringKey() {
-    return getKey(original.getClusteringKey());
+    Optional<Key> clusteringKey = original.getClusteringKey();
+    checkKey(clusteringKey);
+    return clusteringKey;
   }
 
-  private Optional<Key> getKey(Optional<Key> key) {
+  private void checkKey(Optional<Key> key) {
```

Comment:
[minor] This class is immutable. So, I guess executing this kind of check in the constructor makes things simpler (and avoids potential redundant multiple checks). I'm not sure it's worth changing this PR since `getPartitionKey()` and `getPartitionKey()` are deprecated, though.

Reply:
This check ensures that the methods `getPartitionKey()` and `getClusteringKey()` can be used only when the key columns are included in the result. If we enforce this check in the constructor, the result will always need to include the key columns, which is not the expected behavior.